sdk/src/main/java/io/radar/sdk/RadarLocationPermissionManager.kt

```diff
@@ -0,0 +1,116 @@
+            ) { isGranted: Boolean ->
+                if (!isGranted) {
+                    RadarLocationPermissionStatus.savePreviouslyDeniedForeground(context,true)
+                }
+                Radar.sendLocationPermissionStatus(RadarLocationPermissionStatus.initWithStatus(context, activity))
+                // TODO: sync the user's permissions status here
+            }
+
+            requestBackgroundLocationPermissionLauncher = activity.registerForActivityResult(
+                ActivityResultContracts.RequestPermission()
+            ) { isGranted: Boolean ->
+                if (!isGranted) {
+                    RadarLocationPermissionStatus.savePreviouslyDeniedBackground(context, true)
+                }
+                Radar.sendLocationPermissionStatus(RadarLocationPermissionStatus.initWithStatus(context, activity))
+                // TODO: sync the user's permissions status here
+            }
+        } 
+    }
+
+    @RequiresApi(Build.VERSION_CODES.Q)
+    fun requestBackgroundLocationPermission() {
+        if (activity is ComponentActivity) {
+            requestBackgroundLocationPermissionLauncher.launch(Manifest.permission.ACCESS_BACKGROUND_LOCATION)
+            // TODO: sync the user's location permissions action with the their permissions status here
+        }
+    }
+
+    fun requestForegroundLocationPermission() {
+        if (activity is ComponentActivity) {
+            Radar.sendLocationPermissionStatus(RadarLocationPermissionStatus.initWithStatus(context,activity,true))
+            requestForegroundLocationPermissionLauncher.launch(Manifest.permission.ACCESS_FINE_LOCATION)
+            // TODO: sync the user's location permissions action with the their permissions status here
+        }
+    }
+
+    fun openAppSettings() {
+        danglingBackgroundPermissionRequest = true
+        val intent = Intent().apply {
+            action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+            val uri = Uri.fromParts("package", context.packageName, null)
+            data = uri
+            flags = Intent.FLAG_ACTIVITY_NEW_TASK
+        }
+        context.startActivity(intent)
+    }
+
+    fun getLocationPermissionStatus(): RadarLocationPermissionStatus {
+        if (activity is ComponentActivity) {
+            return RadarLocationPermissionStatus.initWithStatus(context, activity)
+        }
+        return RadarLocationPermissionStatus()
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        // do nothing
+    }
+
+    override fun onActivityStopped(p0: Activity) {
+        // do nothing
+    }
+
+    override fun onActivitySaveInstanceState(p0: Activity, p1: Bundle) {
+        // do nothing
+    }
+
+    override fun onActivityDestroyed(p0: Activity) {
+        // do nothing
+    }
+
+    override fun onActivityCreated(p0: Activity, p1: Bundle?) {
+        // do nothing
+    }
+
+    override fun onActivityStarted(p0: Activity) {
+        // do nothing
+    }
```

Comment:
Do we need these overrides if we're doing nothing?

Reply:
Yes, won't compile otherwise.